tests/std/tests/P0896R4_views_transform/test.cpp

```diff
@@ -321,6 +321,50 @@ constexpr bool test_one(Rng&& rng, Expected&& expected) {
     return true;
 }
 
+// Test a function object whose const and non-const versions behave differently
+struct difference_teller {
+    constexpr auto& operator()(auto&& x) noexcept {
+        auto& ref = x;
+        return ref;
+    }
+
+    constexpr auto operator()(auto&& x) const noexcept {
+        return type_identity<decltype(x)>{};
+    }
+};
+
+template <ranges::input_range Rng>
+constexpr void test_difference_on_const_functor(Rng&& rng) {
+    using ranges::transform_view, ranges::input_range, ranges::forward_range, ranges::bidirectional_range,
+        ranges::random_access_range, ranges::iterator_t, ranges::range_reference_t, ranges::range_value_t;
+
+    using V  = views::all_t<Rng>;
+    using TV = transform_view<V, difference_teller>;
+
+    auto r = forward<Rng>(rng) | views::transform(difference_teller{});
+    STATIC_ASSERT(is_same_v<decltype(r), TV>);
+
+    STATIC_ASSERT(is_lvalue_reference_v<range_reference_t<TV>>);
+    if constexpr (input_range<const TV>) {
+        STATIC_ASSERT(is_object_v<range_reference_t<const TV>>);
+        STATIC_ASSERT(!is_same_v<range_value_t<TV>, range_value_t<const TV>>);

```

Comment:
We've established that one of these is a reference type, and the other an object type. They cannot possibly be the same, so this is redundant. (Not worth resetting testing for this alone.)

Reply:
@CaseyCarter The previous lines inspected `range_reference_t`, while this line inspects `range_value_t`, so I believe that they are non-redundant. Am I missing something?

Reply:
No, that's my mistake.

Reply:
I _did_ say it wasn't worth resetting testing =)